manager/src/main/java/com/redhat/service/bridge/manager/vault/AWSVaultServiceImpl.java

```diff
@@ -0,0 +1,139 @@
+package com.redhat.service.bridge.manager.vault;
+
+import java.util.Map;
+import java.util.Optional;
+
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.ApplicationScoped;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.client.builder.AwsClientBuilder;
+import com.amazonaws.secretsmanager.caching.SecretCache;
+import com.amazonaws.secretsmanager.caching.SecretCacheConfiguration;
+import com.amazonaws.services.secretsmanager.AWSSecretsManager;
+import com.amazonaws.services.secretsmanager.AWSSecretsManagerClientBuilder;
+import com.amazonaws.services.secretsmanager.model.CreateSecretRequest;
+import com.amazonaws.services.secretsmanager.model.DeleteSecretRequest;
+import com.amazonaws.services.secretsmanager.model.DeleteSecretResult;
+import com.amazonaws.services.secretsmanager.model.InvalidRequestException;
+import com.amazonaws.services.secretsmanager.model.PutSecretValueRequest;
+import com.amazonaws.services.secretsmanager.model.ResourceExistsException;
+import com.amazonaws.services.secretsmanager.model.ResourceNotFoundException;
+import com.redhat.service.bridge.infra.exceptions.definitions.platform.VaultException;
+import com.redhat.service.bridge.infra.models.EventBridgeSecret;
+
+import io.vertx.core.json.Json;
+
+@ApplicationScoped
+public class AWSVaultServiceImpl implements VaultService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AWSVaultServiceImpl.class);
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.region")
+    String region;
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.endpoint-override")
+    Optional<String> endpoint;
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.access-key-id")
+    String accessKeyId;
+
+    @ConfigProperty(name = "event-bridge.secrets-manager.aws.secret-access-key")
+    String secretAccessKey;
+
+    private AWSSecretsManager awsSecretsManager;
+    private SecretCache secretCache;
+
+    @PostConstruct
+    void init() {
+        AWSCredentialsProvider credentials = new AWSStaticCredentialsProvider(
```

Comment:
@r00ta Have you checked the `quarkus-amazon-services` extensions? https://github.com/quarkiverse/quarkus-amazon-services

It looks like there is support for Secrets Manager in there now. Does it make sense for us to use that instead?